pkg/logs/tag/provider.go

```diff
@@ -10,8 +10,10 @@ import (
 	"sync"
 	"time"
 
+	"github.com/DataDog/datadog-agent/pkg/config"
```

Comment:
I believe `pkg/logs` only accesses the config directly from `pkg/logs/agent.go` or `pkg/logs/config/`. Would make sense to keep this pattern here, but Logs-intake will have a more informed opinion on this.

Reply:
sure thing, I'll ask @DataDog/logs-cold-storage

Reply:
this is correct, most of the times we inject dependencies for unit-tests and avoid accessing datadog config directly